File: web3rpc/rpc-specs/code-samples/python/debug/vMTracing/traceCall.py
from web3 import Web3
from web3py_ext import extend

host = "https://api.baobab.klaytn.net:8651"

tracerCallObject = {"to":"0x46eda75e7ca73cb1c2f83c3927211655420dbc44","data":"0x3fb5c1cb00000000000000000000000000000000000000000000000000000000000003e7"};
blockNumber = "latest"
options = {"tracer":"revertTracer"}
w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.trace_call(tracerCallObject, blockNumber, options)

print(debug_response)

File: web3rpc/rpc-specs/code-samples/javascript/debug/vMTracing/traceCall.js
const { JsonRpcProvider } = require("@klaytn/ethers-ext");

(async () => {
  const provider = new JsonRpcProvider("https://public-en-baobab.klaytn.net");
  const tracerCallObject = {
    to: "0x46eda75e7ca73cb1c2f83c3927211655420dbc44",
    data: "0x3fb5c1cb00000000000000000000000000000000000000000000000000000000000003e7",
  };
  const blockNumber = "latest";
  const options = { tracer: "revertTracer" };

  // The full list of JSON-RPC is available at:
  // https://docs.klaytn.foundation/content/dapp/json-rpc/api-references
  const data = await provider.debug.traceCall(
    tracerCallObject,
    blockNumber,
    options,
  );
  console.log("Trace call", data);
})();
